pandas/core/ops.py

```diff
@@ -775,6 +775,12 @@ def wrapper(left, right, name=name, na_op=na_op):
             res_name = left.name
 
         result = wrap_results(safe_na_op(lvalues, rvalues))
+        try:
```

Comment:
I would do this in _construct_result and simply pass in ``name = res_name or getattr(result, 'name', None)``

Reply:
Good call.